network/src/network.rs

```diff
@@ -549,7 +570,7 @@ impl EventHandler {
     }
 }
 
-impl ServiceHandle for EventHandler {
+impl<T: ExitHandler> ServiceHandle for EventHandler<T> {
```

Comment:
Since `ExitHandler` is only called once, there's no performance issue. I think we can use trait box instead of generic to make the code a bit simpler.

Reply:
> Since `ExitHandler` is only called once, there's no performance issue, we can use trait box instead of generic to make the code a bit simpler.

I agree with ian.

Reply:
And I think it is better to define `ExitHandler` and `DefaultExitHandler` in `ckb-util` (`ckb-util` is small enough).